lib/appsignal/hooks/sidekiq.rb

```diff
@@ -20,6 +23,60 @@ def install
       end
     end
 
+    class SidekiqProbe
+      def initialize
+        @cache = {}
+      end
+
+      def call
+        stats = ::Sidekiq::Stats.new
+        redis_info = ::Sidekiq.redis_info
+        gauge "worker_count", stats.workers_size
+        gauge "process_count", stats.processes_size
+        gauge "connection_count", redis_info.fetch("connected_clients")
+        gauge "memory_usage", redis_info.fetch("used_memory")
+        gauge "memory_usage_rss", redis_info.fetch("used_memory_rss")
+        gauge_delta :jobs_processed, "job_count", stats.processed,
+          :status => :processed
+        gauge_delta :jobs_failed, "job_count", stats.failed, :status => :failed
+        gauge "job_count", stats.retry_size, :status => :retry_queue
```

Comment:
What is the reason that we don't calculate the delta for retry, scheduled and enqueued?

Reply:
Sidekiq gives you multiple kinds of data through the Sidekiq::Stats class.
We don't calculate deltas for "retry", "scheduled" and "enqueued",  because those metrics are "queue sizes". They are already values we can safely send as gauges. The values go up and down based on how the length of the specific queue.

The "processed", "failed", "dead" Sidekiq stats are counters (they only go up) we store as gauges by calculating the differences of each counter per probe call.

Reply:
Ah yeah that makes sense.